java/vector/src/main/java/org/apache/arrow/vector/ipc/ReadChannel.java

```diff
@@ -53,11 +53,14 @@ public long bytesRead() {
    * @throws IOException if nit enough bytes left to read
    */
   public int readFully(ByteBuffer buffer) throws IOException {
-    LOGGER.debug("Reading buffer with size: {}", buffer.remaining());
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Reading buffer with size: {}", buffer.remaining());
```

Comment:
Is `buffer.remaining()` an expensive enough operation to warrant an explicit log level check?

Reply:
Good question. 
IMO, buffer.remaining() is not expensive. It is a generally considered as common practice to wrap debug log in an if statements (e.g. see https://stackoverflow.com/questions/10428447/log-debug-enabled-check-in-java). The benefits: 1) avoid expensive operations when logging; 2) since the if statement always takes the same branch, JIT can make some optimizations here (maybe removing the if statement altogether)